mne/annotations.py

```diff
@@ -857,6 +857,36 @@ def _select_annotations_based_on_description(descriptions, event_id, regexp):
     return event_sel, event_id_
 
 
+def _select_events_based_on_id(events, event_desc):
+    """Get a collection of events and returns index of selected."""
+    event_desc_ = dict()
+    dropped = []
+    # Iterate over the sorted descriptions so that the Counter mapping
+    # is slightly less arbitrary
+    for e in events:
+        if e[2] in event_desc_:
+            continue
```

Comment:
I would just do:
```
event_nums = events[:, np.unique(events[:, 2], return_index=True)[1]]
```
and iterate over `for e in event_nums`, then you don't need this conditional (you process each int exactly once)